tests/ops/qubit/test_special_unitary.py

```diff
@@ -93,6 +96,346 @@ def test_pauli_basis_strings(self, n):
+        from jax import numpy as jnp
+
+        n = 1
+        fn = (
+            jax.jit(self.get_one_parameter_generators, static_argnums=[1, 2])
+            if use_jit
+            else self.get_one_parameter_generators
+        )
+        basis = pauli_basis_matrices(n)
+        d = 4**n - 1
+        for i, (theta, pauli_mat) in enumerate(zip(jnp.eye(d), basis)):
+            Omegas = fn(theta, n, "jax")
+            assert Omegas.shape == (d, 2**n, 2**n)
+            assert all(jnp.allclose(O.conj().T, -O) for O in Omegas)
+            assert jnp.allclose(Omegas[i], 1j * pauli_mat)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("n", [1, 2, 3])
+    def test_tf(self, n):
+        """Test that generators are computed correctly in Tensorflow."""
+        import tensorflow as tf
+
+        np.random.seed(14521)
+        d = 4**n - 1
+        theta = tf.Variable(np.random.random(d))
+        Omegas = self.get_one_parameter_generators(theta, n, "tf")
+        assert Omegas.shape == (d, 2**n, 2**n)
+        assert all(qml.math.allclose(qml.math.conj(qml.math.T(O)), -O) for O in Omegas)
+
+    @pytest.mark.tf
+    def test_tf_pauli_generated(self):
+        """Test that generators match Pauli words."""
+        import tensorflow as tf
+
+        n = 1
+        basis = pauli_basis_matrices(n)
+        d = 4**n - 1
+        for i, (theta, pauli_mat) in enumerate(zip(np.eye(d), basis)):
+            theta = tf.Variable(theta)
+            Omegas = self.get_one_parameter_generators(theta, n, "tf")
+            assert Omegas.shape == (d, 2**n, 2**n)
+            assert all(qml.math.allclose(qml.math.conj(qml.math.T(O)), -O) for O in Omegas)
+            assert qml.math.allclose(Omegas[i], 1j * pauli_mat)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize("n", [1, 2, 3])
+    def test_torch(self, n):
+        """Test that generators are computed correctly in Torch."""
+        import torch
+
+        np.random.seed(14521)
+        d = 4**n - 1
+        theta = torch.tensor(np.random.random(d), requires_grad=True)
+        Omegas = self.get_one_parameter_generators(theta, n, "torch")
+        assert Omegas.shape == (d, 2**n, 2**n)
+        assert all(qml.math.allclose(qml.math.conj(qml.math.T(O)), -O) for O in Omegas)
+
+    @pytest.mark.torch
+    def test_torch_pauli_generated(self):
+        """Test that generators match Pauli words."""
+        import torch
+
+        n = 1
+        basis = pauli_basis_matrices(n)
+        d = 4**n - 1
+        for i, (theta, pauli_mat) in enumerate(zip(torch.eye(d, requires_grad=True), basis)):
+            Omegas = self.get_one_parameter_generators(theta, n, "torch")
+            assert Omegas.shape == (d, 2**n, 2**n)
+            assert all(
+                qml.math.allclose(qml.math.conj(qml.math.T(O)), -O, atol=3e-8) for O in Omegas
+            )
+            assert qml.math.allclose(Omegas[i], 1j * pauli_mat)
+
+    # Autograd does not support differentiating expm.
+    @pytest.mark.xfail
```

Comment:
Do we need this test? We are already testing for the `raise error` in the next test.

Reply:
Ah, good question. I included it originally in case we get `expm` to work in a differentiable manner. I'll delete it, it's easy to add back in once we have that feature.